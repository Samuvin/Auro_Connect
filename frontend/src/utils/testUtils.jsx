import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Create a custom render function that includes providers
export const renderWithProviders = (ui, options = {}) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  const AllTheProviders = ({ children }) => {
    return (
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

// Mock user data for testing
export const mockUser = {
  _id: '1',
  name: 'John Doe',
  username: 'johndoe',
  email: 'john@example.com',
  profilePicture: '/avatar.png',
  headline: 'Software Developer',
  connections: ['2', '3'],
  location: 'New York, NY'
};

// Mock post data for testing
export const mockPost = {
  _id: '1',
  content: 'This is a test post',
  author: mockUser,
  likes: ['2', '3'],
  comments: [
    {
      _id: '1',
      content: 'Great post!',
      user: mockUser,
      createdAt: new Date().toISOString()
    }
  ],
  image: 'https://example.com/image.jpg',
  createdAt: new Date().toISOString()
}; 