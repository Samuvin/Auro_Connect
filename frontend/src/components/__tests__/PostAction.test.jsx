import { screen, fireEvent } from '@testing-library/react';
import PostAction from '../PostAction';
import { renderWithProviders } from '../../utils/testUtils';

describe('PostAction', () => {
  const mockOnClick = jest.fn();
  const mockIcon = <span data-testid="mock-icon">👍</span>;

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders with icon and text', () => {
    renderWithProviders(
      <PostAction 
        icon={mockIcon} 
        text="Like" 
        onClick={mockOnClick} 
      />
    );
    
    expect(screen.getByTestId('mock-icon')).toBeInTheDocument();
    expect(screen.getByText('Like')).toBeInTheDocument();
  });

  test('calls onClick when button is clicked', () => {
    renderWithProviders(
      <PostAction 
        icon={mockIcon} 
        text="Like" 
        onClick={mockOnClick} 
      />
    );
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  test('renders only icon when text is not provided', () => {
    renderWithProviders(
      <PostAction 
        icon={mockIcon} 
        onClick={mockOnClick} 
      />
    );
    
    expect(screen.getByTestId('mock-icon')).toBeInTheDocument();
    expect(screen.queryByText('Like')).not.toBeInTheDocument();
  });

  test('renders with different icons and texts', () => {
    const commentIcon = <span data-testid="comment-icon">💬</span>;
    
    renderWithProviders(
      <PostAction 
        icon={commentIcon} 
        text="Comment" 
        onClick={mockOnClick} 
      />
    );
    
    expect(screen.getByTestId('comment-icon')).toBeInTheDocument();
    expect(screen.getByText('Comment')).toBeInTheDocument();
  });

  test('has correct CSS classes for responsive design', () => {
    renderWithProviders(
      <PostAction 
        icon={mockIcon} 
        text="Share" 
        onClick={mockOnClick} 
      />
    );
    
    const button = screen.getByRole('button');
    expect(button).toHaveClass('flex', 'items-center');
    
    const textSpan = screen.getByText('Share');
    expect(textSpan).toHaveClass('hidden', 'sm:inline');
  });

  test('renders without onClick handler', () => {
    renderWithProviders(
      <PostAction icon={mockIcon} text="Like" />
    );
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    // Should not throw error when onClick is undefined
    expect(button).toBeInTheDocument();
  });

  // Snapshot Tests
  test('matches snapshot for post action with icon and text', () => {
    const { container } = renderWithProviders(
      <PostAction 
        icon={mockIcon} 
        text="Like" 
        onClick={mockOnClick} 
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for post action with icon only', () => {
    const { container } = renderWithProviders(
      <PostAction 
        icon={mockIcon} 
        onClick={mockOnClick} 
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for post action with comment icon', () => {
    const commentIcon = <span data-testid="comment-icon">💬</span>;
    const { container } = renderWithProviders(
      <PostAction 
        icon={commentIcon} 
        text="Comment" 
        onClick={mockOnClick} 
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for post action without onClick handler', () => {
    const { container } = renderWithProviders(
      <PostAction icon={mockIcon} text="Like" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
}); 