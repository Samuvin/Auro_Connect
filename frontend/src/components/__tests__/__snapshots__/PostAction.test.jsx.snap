// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PostAction matches snapshot for post action with comment icon 1`] = `
<button
  class="flex items-center"
>
  <span
    class="mr-1"
  >
    <span
      data-testid="comment-icon"
    >
      💬
    </span>
  </span>
  <span
    class="hidden sm:inline"
  >
    Comment
  </span>
</button>
`;

exports[`PostAction matches snapshot for post action with icon and text 1`] = `
<button
  class="flex items-center"
>
  <span
    class="mr-1"
  >
    <span
      data-testid="mock-icon"
    >
      👍
    </span>
  </span>
  <span
    class="hidden sm:inline"
  >
    Like
  </span>
</button>
`;

exports[`PostAction matches snapshot for post action with icon only 1`] = `
<button
  class="flex items-center"
>
  <span
    class="mr-1"
  >
    <span
      data-testid="mock-icon"
    >
      👍
    </span>
  </span>
  <span
    class="hidden sm:inline"
  />
</button>
`;

exports[`PostAction matches snapshot for post action without onClick handler 1`] = `
<button
  class="flex items-center"
>
  <span
    class="mr-1"
  >
    <span
      data-testid="mock-icon"
    >
      👍
    </span>
  </span>
  <span
    class="hidden sm:inline"
  >
    Like
  </span>
</button>
`;
